src/main/java/seedu/address/logic/commands/ViewCommand.java

```diff
@@ -48,6 +50,10 @@ public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Task> lastShownTaskList = model.getFilteredTaskList();
 
+        List<Person> lastShownPersonList = new ArrayList<>();
```

Comment:
Instead of this, is it possible to just throw a CommandException(Messages.NO_CONTACTS_ASSIGNED) if listSize == 0?

Reply:
`lastShownPersonList` is used to hold a copy of the previous contact list displayed to the user . This is then converted to a predicate to be used on line 73

Ngl, I find it weird that I have to do this:
`model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);` <--Reset the contact list
`model.updateFilteredPersonList(originalStatePred);` <-- Make no changes to the contact list


However, this is because of line 67:
`model.updateFilteredPersonList(predicate);` <-- If task is empty, this will "force" the UI to display the contact as empty.

I have to do `model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);` first, instead of just doing `model.updateFilteredPersonList(originalStatePred);` because if I were to only use the latter, it will see that the list is "empty" and will continue to show empty. So the formal is used to populate the list.

Reply:
> Instead of this, is it possible to just throw a `CommandException(Messages.NO_CONTACTS_ASSIGNED)` if `listSize == 0`?

> Instead of this, is it possible to just throw a `CommandException(Messages.NO_CONTACTS_ASSIGNED)` if `listSize == 0`?

Reply:
`lastShownPersonList` is used to hold a copy of the previous contact list displayed to the user . This is then converted to a predicate to be used on line 73

Ngl, I find it weird that I have to do this:
`model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);` <--Reset the contact list
`model.updateFilteredPersonList(originalStatePred);` <-- Make no changes to the contact list


However, this is because of line 67:
`model.updateFilteredPersonList(predicate);` <-- If task is empty, this will "force" the UI to display the contact as empty.

I have to do `model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);` first, instead of just doing `model.updateFilteredPersonList(originalStatePred);` because if I were to only use the latter, it will see that the list is "empty" and will continue to show empty. So the formal is used to populate the list.